src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssSkuProfileVmSizeCommand.cs

```diff
@@ -0,0 +1,116 @@
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet(VerbsCommon.Add, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VmssSkuProfileVMSize", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSVirtualMachineScaleSet))]
+    public partial class AddAzureRmVmssSkuProfileVmSizeCommand : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        public PSVirtualMachineScaleSet VirtualMachineScaleSet { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true)]
+        public string VmSize { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true)]
+        public int? Rank { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true)]
+        public string AllocationStrategy { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (ShouldProcess("VirtualMachineScaleSet", "Add"))
+            {
+                Run();
+            }
+        }
+
+        private void Run()
+        {
+            if (this.VirtualMachineScaleSet.SkuProfile == null)
+            {
+                this.VirtualMachineScaleSet.SkuProfile = new SkuProfile();
+            }
+
+            if (this.VirtualMachineScaleSet.SkuProfile.VmSizes  == null) 
+            {
+                this.VirtualMachineScaleSet.SkuProfile.VmSizes = new List<SkuProfileVMSize>();
+            }
+
+            if (this.IsParameterBound(c => c.VmSize))
+            {
+                bool isVmSizeAlreadyPresent = this.VirtualMachineScaleSet.SkuProfile.VmSizes.Any(vmSize => vmSize.Name.Equals(this.VmSize, StringComparison.OrdinalIgnoreCase));
+                if (isVmSizeAlreadyPresent)
+                {
+                    throw new ArgumentException($"The VM size '{this.VmSize}' is already present in the SkuProfile.");
+                }
+
+                this.VirtualMachineScaleSet.SkuProfile.VmSizes.Add(new SkuProfileVMSize()
+                {
+                    Name = this.VmSize,
+                    Rank = this.Rank
+                });
+            }
+
+            if (this.IsParameterBound(c => c.AllocationStrategy))
+            {
+                this.VirtualMachineScaleSet.SkuProfile.AllocationStrategy = this.AllocationStrategy;
+            }
+            else if (this.VirtualMachineScaleSet.SkuProfile.AllocationStrategy == null)
+            {
+                if (this.IsParameterBound(c => c.Rank))
```

Comment:
this type of defaulting should happen in the service side. We want to avoid adding this kind of logic at the PowerShell level